### PR TITLE
[PATCH banner] adds backgroundColor prop to Banner

### PR DESCRIPTION
**Pull request by kierangillen**
- Adds backgroundColor to banner component for transitions
- Possibly needs some animation added as well, need to check with design team

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-palette* ready!

Built with commit 7a01a2438206914bc8cd6fdedf6acaafd2074f28

https://deploy-preview-139--artsy-palette.netlify.com

File: src/elements/Banner/Banner.tsx [hunk 1]
```diff
@@ -34,6 +34,7 @@ const CloseButton = ({ onClick }) => {
 export interface BannerProps {
   dismissable: boolean
   message: string
+  backgroundColor?: string
```
Comment (damassi) [inline src/elements/Banner/Banner.tsx]: Would you mind updating [the test](https://github.com/artsy/palette/blob/7a01a2438206914bc8cd6fdedf6acaafd2074f28/src/elements/Banner/Banner.test.tsx) with this new API option?

**Comment by jonallured** [comment]
Nice to see this little component learning some new tricks! 👍 

Here's what I'm thinking:

* Add a case to this file: docs/Banner.mdx maybe called something like "Status with Custom Background" and then demonstrate overriding the default.
* Since the default color is that red, let's promote that up into the `defaultProps` bit and that'll make the property required not optional.
* Pass `this.props.backgroundColor` directly to `Box` rather than creating a local for it in the render.

How does that sound to you?

File: src/elements/Banner/Banner.tsx [hunk 2]
```diff
@@ -23,6 +23,10 @@ const Target = styled.div`
   }
 `
 
+const Wrapper = styled(Box)`
+  transition: background-color 250ms linear;
```
Comment (starsirius) [inline src/elements/Banner/Banner.tsx]: Do we want to transition the background color? I might miss the conversations, but this feels a new UI pattern worth checking in with design.

**Comment by starsirius** [comment]
As @damassi mentioned, let's add a test in https://github.com/artsy/palette/blob/master/src/elements/Banner/Banner.test.tsx and test the option!

**Comment by starsirius** [comment]
Nice one! 🥇

**Pull request merged**